packages/ui/src/components/primitives/Message.tsx

```diff
@@ -24,15 +27,74 @@ export function useMessageContext() {
 	return context;
 }
 
+const CopyIcon = () => {
+	return (
+		<svg
+			xmlns="http://www.w3.org/2000/svg"
+			fill="none"
+			viewBox="0 0 24 24"
+			strokeWidth={1.5}
+			stroke="currentColor"
+			className="h-4 w-4"
+		>
+			<path
+				strokeLinecap="round"
+				strokeLinejoin="round"
+				d="M15.666 3.888A2.25 2.25 0 0013.5 2.25h-3c-1.03 0-1.9.693-2.166 1.638m7.332 0c.055.194.084.4.084.612v0a.75.75 0 01-.75.75H9a.75.75 0 01-.75-.75v0c0-.212.03-.418.084-.612m7.332 0c.646.049 1.288.11 1.927.184 1.1.128 1.907 1.077 1.907 2.185V19.5a2.25 2.25 0 01-2.25 2.25H6.75A2.25 2.25 0 014.5 19.5V6.257c0-1.108.806-2.057 1.907-2.185a48.208 48.208 0 011.927-.184"
+			/>
+		</svg>
+	);
+};
+
+const ExternalLinkIcon = () => {
+	return (
+		<svg
+			xmlns="http://www.w3.org/2000/svg"
+			fill="none"
+			viewBox="0 0 24 24"
+			strokeWidth={1.5}
+			stroke="currentColor"
+			className="h-4 w-4"
+		>
+			<path
+				strokeLinecap="round"
+				strokeLinejoin="round"
+				d="M13.5 6H5.25A2.25 2.25 0 003 8.25v10.5A2.25 2.25 0 005.25 21h10.5A2.25 2.25 0 0018 18.75V10.5m-10.5 6L21 3m0 0h-5.25M21 3v5.25"
+			/>
+		</svg>
+	);
+};
+
 export const MessageAuthorArea = () => {
+	const [copyStatus, copyToClipboard] = useCopyToClipboard();
 	const { message } = useMessageContext();
+
 	return (
 		<div className="flex w-full min-w-0 gap-2">
 			{/* TODO: sort out responsive styling */}
 			<div className="flex w-full flex-row items-center gap-2 font-body text-lg text-black/[.7] dark:text-white/[.47]">
 				<DiscordAvatar user={message.author} size="sm" />
 				<span className="mr-1">{message.author.name}</span>
-				<span className="ml-auto">{getSnowflakeUTCDate(message.id)}</span>
+				<div className="mr-4 ml-auto flex flex-row gap-2">
+					<LinkButton
+						href={`https://discord.com/${message.serverId}/${message.id}`}
+						className="p-1"
+					>
+						<ExternalLinkIcon />
+					</LinkButton>
+
+					<Button
+						className="p-1"
+						onClick={() =>
+							copyToClipboard(
+								`https://discord.com/${message.serverId}/${message.id}`,
+							)
+						}
+					>
+						<CopyIcon />
```

Comment:
Copy URL isn't needed, if we were to have it the place that would make sense for it is in the headline for the message result page

Reply:
Removed!